state/stateChangesCollector.go

```diff
@@ -0,0 +1,67 @@
+package state
+
+// DataTrieChange represents a change in the data trie
+type DataTrieChange struct {
+	Key []byte
+	Val []byte
+}
+
+// StateChangeDTO is used to collect state changes
+type StateChangeDTO struct {
+	MainTrieKey     []byte
+	MainTrieVal     []byte
+	DataTrieChanges []DataTrieChange
+}
+
+// StateChangesForTx is used to collect state changes for a transaction hash
+type StateChangesForTx struct {
+	TxHash       []byte
+	StateChanges []StateChangeDTO
+}
+
+type stateChangesCollector struct {
+	stateChanges      []StateChangeDTO
+	stateChangesForTx []StateChangesForTx
+}
+
+// NewStateChangesCollector creates a new StateChangesCollector
+func NewStateChangesCollector() *stateChangesCollector {
+	return &stateChangesCollector{
+		stateChanges: []StateChangeDTO{},
+	}
+}
+
+// AddStateChange adds a new state change to the collector
+func (scc *stateChangesCollector) AddStateChange(stateChange StateChangeDTO) {
+	scc.stateChanges = append(scc.stateChanges, stateChange)
+}
+
+// GetStateChanges returns the accumulated state changes
+func (scc *stateChangesCollector) GetStateChanges() []StateChangesForTx {
+	if len(scc.stateChanges) > 0 {
+		scc.AddTxHashToCollectedStateChanges([]byte{})
```

Comment:
so if `GetStateChanges` will be called after several `AddStateChange` without `AddTxHashToCollectedStateChanges` those state changes will remain with empty tx hash, does it help to reference state changes without an identifier?

Reply:
This was addressed in your PR.